data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/GenesisTimeHandler.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
+
+import io.javalin.plugin.json.JavalinJson;
+import java.util.HashMap;
+import java.util.Map;
+import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+
+public class GenesisTimeHandler implements BeaconRestApiHandler {
+
+  private String path = "/node/genesis_time";
+  private String returnString;
+
+  public GenesisTimeHandler(ChainStorageClient client) {
+    Map<String, Long> jsonObject = new HashMap<>();
+    jsonObject.put("genesis_time", client.getGenesisTime().longValue());
+    returnString = JavalinJson.toJson(jsonObject);
```

Comment:
Is the genesis time guaranteed to have been inited by the time we create this handler?  What happens when we're running pre-genesis event on a real network?

Reply:
Good question. Currently, we get the genesis time from config.toml and set it to the chainStorage.genesisTime, (regardless of the startup method?), so we always have a genesisTime by the time any API handler is initialized. How would we want them to work on in the perfect scenario?